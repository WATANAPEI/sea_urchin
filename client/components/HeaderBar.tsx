import React from "react";
import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1
    }
  })
);

interface Props {
  text?: string;
}

function HeaderBar({ text }: Props) {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <AppBar color="default" position="static">
        <Toolbar>
          <Typography variant="h6" color="inherit">
            {text}
          </Typography>
        </Toolbar>
      </AppBar>
    </div>
  );
}

export default HeaderBar;
