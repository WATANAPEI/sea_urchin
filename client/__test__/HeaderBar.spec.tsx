import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
// @ts-ignore
import HeaderBar from "../components/HeaderBar.tsx";
import AppBar from "@material-ui/core/AppBar";

describe("HeaderBar", () => {
  it("display the received text", () => {
    const testStr = "this is test";
    const wrapper = createShallow()(<HeaderBar text={testStr} />);
    expect(wrapper.text()).toEqual(testStr);
    expect(wrapper).toMatchSnapshot();
  });

  it("has an AppBar", () => {
    const testStr = "this is test";
    const wrapper = createShallow()(<HeaderBar text={testStr} />);
    expect(wrapper.find(AppBar)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
