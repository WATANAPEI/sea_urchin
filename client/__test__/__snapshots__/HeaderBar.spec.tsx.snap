// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBar display the received text 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(AppBar))
    color="default"
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Typography))
        color="inherit"
        variant="h6"
      >
        this is test
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;

exports[`HeaderBar has an AppBar 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(AppBar))
    color="default"
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Typography))
        color="inherit"
        variant="h6"
      >
        this is test
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;
